src/dxvk/rtx_render/rtx_types.cpp

```diff
@@ -128,39 +128,46 @@ namespace dxvk {
     }
   }
 
+  static bool wasCategorySetViaD3D(InstanceCategories category, uint32_t categoryFlags) {
+    return categoryFlags & (1 << static_cast<uint32_t>(category));
+  }
+
   void DrawCallState::setupCategoriesForTexture() {
     // TODO (REMIX-231): It would probably be much more efficient to use a map of texture hash to category flags, rather
     //                   than doing N lookups per texture hash for each category.
     const XXH64_hash_t& textureHash = materialData.getColorTexture().getImageHash();
 
-    setCategory(InstanceCategories::WorldUI, lookupHash(RtxOptions::worldSpaceUiTextures(), textureHash));
-    setCategory(InstanceCategories::WorldMatte, lookupHash(RtxOptions::worldSpaceUiBackgroundTextures(), textureHash));
+    const auto& d3dCategoryEnabled = RtxOptions::Get()->useUnusedRenderstates();
+    const auto& d3dCategoryFlag = materialData.remixTextureCategoryFlagsFromD3D;
 
-    setCategory(InstanceCategories::Ignore, lookupHash(RtxOptions::ignoreTextures(), textureHash));
-    setCategory(InstanceCategories::IgnoreLights, lookupHash(RtxOptions::ignoreLights(), textureHash));
-    setCategory(InstanceCategories::IgnoreAntiCulling, lookupHash(RtxOptions::antiCullingTextures(), textureHash));
-    setCategory(InstanceCategories::IgnoreMotionBlur, lookupHash(RtxOptions::motionBlurMaskOutTextures(), textureHash));
-    setCategory(InstanceCategories::IgnoreOpacityMicromap, lookupHash(RtxOptions::opacityMicromapIgnoreTextures(), textureHash) || isUsingRaytracedRenderTarget);
-    setCategory(InstanceCategories::IgnoreAlphaChannel, lookupHash(RtxOptions::ignoreAlphaOnTextures(), textureHash));
-    setCategory(InstanceCategories::IgnoreBakedLighting, lookupHash(RtxOptions::ignoreBakedLightingTextures(), textureHash));
+    setCategory(InstanceCategories::WorldUI, lookupHash(RtxOptions::worldSpaceUiTextures(), textureHash) || (d3dCategoryEnabled & wasCategorySetViaD3D(InstanceCategories::WorldUI, d3dCategoryFlag)));
+    setCategory(InstanceCategories::WorldMatte, lookupHash(RtxOptions::worldSpaceUiBackgroundTextures(), textureHash) || (d3dCategoryEnabled & wasCategorySetViaD3D(InstanceCategories::WorldMatte, d3dCategoryFlag)));
```

Comment:
using a single & is doing a bitwise and between `d3dCategoryEnabled` and the bool returned by `wasCategorySetViaD3D`, which probably isn't what you meant to do.  It should be `&&`, but I think this entire set of modifications is actually not needed.

Instead, you should just make `materialData.remixTextureCategoryFlagsFromD3D` a `CategoryFlags`, and then at the top of this function just do

```
if (RtxOptions::Get()->useUnusedRenderstates()) {
  categories = materialData.remixTextureCategoryFlagsFromD3D;
}
setCategory(...
```

Reply:
Indeed! That was not intended 🤨
Did not know about CategoryFlags. That makes the code much cleaner!

`remixTextureCategoryFlagsFromD3D` is still a uint32_t because I could't use `CategoryFlags` as a type in `rtx_materials.h` and I would have to include `rtx_types.h`.
-> I'm now directly assigning `categories` using CategoryFlags in `rtx_types.cpp`